### DISCUSSION TITLE
How to communicate across stores.

### DISCUSSION THREAD

**Discussion by KyryloSWAN** [mobxjs/mobx-state-tree#1993] [Q&A] [answered]
I need to get property value from one store and use it in another. I can't find any documentation about this problem. How I can do this?

**Comment by coolsoftwaretyler** [comment, marked as answer]
Hey @KyryloSWAN - sorry it's been so long without a reply. Perhaps you were able to figure it out yourself, but I have some thoughts for you, or other folks with a similar question. 

Assuming you are talking about _sibling stores_, such that there is a parent store, a child store A, and a child store B, you can think about it in two ways:

1. One sibling, say child store A, uses [getParent](https://mobx-state-tree.js.org/API/#getparent), or [getParentOfType](https://mobx-state-tree.js.org/API/#getparentoftype) to get a reference to the parent store, and then asks the parent store to get a value from child store B.
2. Your consuming application (such as a React component, or whatever else you are using alongside MST) requests the data from the parent store, and then provides it to the child store. So instead of child store A ever having to think about its parent, it has some method on it that takes an input, and you have your component get that input from the parent (or even directly from child store B). 

There are tradeoffs between these approaches, but I think they're both valid and common.

Does that answer your question? I am happy to provide more details or explore tradeoffs as needed.

**Reply by KyryloSWAN** [reply]
Thank you for your answer, @coolsoftwaretyler!
You right, I've already resolved my problem with communication between different parts of store by using getParent and views:
```
.views((self) => ({
    get token(): string {
      return getParent<typeof RootStore>(self).token;
    },
  }))
```
